src/query/storages/fuse/src/operations/mutation/mutation_aggregator.rs

```diff
@@ -0,0 +1,289 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::sync::Arc;
+use std::time::Instant;
+
+use common_base::runtime::execute_futures_in_parallel;
+use common_catalog::table_context::TableContext;
+use common_exception::Result;
+use common_expression::BlockMetaInfoDowncast;
+use common_expression::BlockThresholds;
+use common_expression::DataBlock;
+use common_expression::TableSchemaRef;
+use common_pipeline_transforms::processors::transforms::AsyncAccumulatingTransform;
+use opendal::Operator;
+use storages_common_table_meta::meta::BlockMeta;
+use storages_common_table_meta::meta::Location;
+use storages_common_table_meta::meta::SegmentInfo;
+use storages_common_table_meta::meta::Statistics;
+use storages_common_table_meta::meta::Versioned;
+use tracing::info;
+
+use crate::io::SegmentsIO;
+use crate::io::SerializedSegment;
+use crate::io::TableMetaLocationGenerator;
+use crate::operations::mutation::AbortOperation;
+use crate::operations::mutation::Mutation;
+use crate::operations::mutation::MutationSinkMeta;
+use crate::operations::mutation::MutationTransformMeta;
+use crate::statistics::reducers::deduct_statistics_mut;
+use crate::statistics::reducers::merge_statistics_mut;
+use crate::statistics::reducers::reduce_block_metas;
+
+type MutationMap = HashMap<usize, (Vec<(usize, Arc<BlockMeta>)>, Vec<usize>)>;
```

Comment:
can define `(Vec<(usize, Arc<BlockMeta>)>, Vec<usize>)` as a type to be more clearly?